string address
bool enablePub
bool enableSub
bool shouldProcess